witness/src/Dscp/Network/Wrapped.hs

```diff
@@ -109,104 +114,75 @@ subType = Subscription $ BS8.pack $ show $ getMsgTag @SubK @d
 fromSubType :: Subscription -> Maybe Natural
 fromSubType (Subscription bs) = readMaybe (BS8.unpack bs)
 
----------------------------------------------------------------------------
--- Listeners
+----------------------------------------------------------------------------
+-- Workers
 ----------------------------------------------------------------------------
 
--- Listeners are supposed to have one dispatcher only (and one receive
--- block only). When new messages come, response callback thread is
--- forked and this thread is allowed to send multiple messages
--- (directly, or publish).
-
-data Listener m = Listener
-    { lId       :: !ListenerId
-      -- ^ Listener id, should be unique.
-    , lMsgTypes :: !(Set MsgType)
-      -- ^ Message types listener is supposed to receive.
-    , lAction   :: !(ListenerEnv NetTag -> m Void)
-      -- ^ Listener's action. Should never terminate.
+data Worker m = forall pre. Worker
+    { wId        :: !ClientId
+      -- ^ Worker's identity.
+    , wBootstrap :: !(m pre)
+      -- ^ Initialize necessary context. Fail here or never.
+    , wAction    :: !(pre -> m Void)
+      -- ^ Worker's action. Should never end.
+    , wRecovery  :: !(RetryPolicyM m)
```

Comment:
Why don't we have `StrictData` already?

Reply:
Nice, this module is a good place for that.